src/main/java/org/terasology/simpleFarming/systems/VineAuthoritySystem.java

```diff
@@ -118,14 +115,32 @@ public void onVinePlanted(OnSeedPlanted event, EntityRef definitionEntity, VineD
      */
     @ReceiveEvent
     public void onVineGrowth(DelayedActionTriggeredEvent event, EntityRef root, VineNodeComponent nodeComponent, VineDefinitionComponent vineComponent) {
-        if (nodeComponent.height != -1) {
-            if (nodeComponent.height < 20) {
+        doGrowVine(root, nodeComponent, vineComponent);
+    }
+
+    private void doGrowVine(EntityRef root, VineNodeComponent nodeComponent, VineDefinitionComponent vineComponent) {
+        if (nodeComponent.length != -1) {
+            if (nodeComponent.length < 20) {
```

Comment:
I noticed that you added a `maxLength` to the `VineDefinitionComponent` but it's not used anywhere. This 20 is the vine's length so replacing it with `maxLength` will implement it.

Reply:
Hah,  I didnt quite finish implementing that one.  Doh.

Also,  @flo and I made some long vines yesterday with the growthcan...  much fun was had!